packages/server/modules/shared/authz.ts

```diff
@@ -0,0 +1,293 @@
+import { Scopes } from '@/modules/core/helpers/mainConstants'
+import { getRoles } from '@/modules/shared'
+import { getStream } from '@/modules/core/services/streams'
+
+import {
+  BaseError,
+  ForbiddenError as FE,
+  UnauthorizedError as UE,
+  ContextError as CE,
+  BadRequestError as BRE
+} from '@/modules/shared/errors'
+// import { getbAllRoles } from '../core/services/generic'
+
+interface AuthResult {
+  authorized: boolean
+}
+
+interface AuthFailedResult extends AuthResult {
+  authorized: false
+  error: BaseError | null
+  fatal?: boolean
+}
+
+interface Stream {
+  role?: StreamRoles
+  isPublic: boolean
+  allowPublicComments: boolean
+}
+
+interface AuthContext {
```

Comment:
maybe some other ones as well, but I know that this AuthContext type already has a JSDoc typedef in `modules/shared/index.js` (AuthContextPart)

to make things more DRY, you should delete the JSDoc Typedef, and use this new TS type instead. you can import & use TS types in JSDoc comments as well, through the dynamic import syntax.